Celeste.Mod.mm/Patches/MapData.cs

```diff
@@ -45,10 +47,18 @@ public patch_MapData(AreaKey area)
             DetectedCassette = false;
             DetectedStrawberriesIncludingUntracked = 0;
 
+            if (levelsByName is null) {
```

Comment:
don't think this should ever be null since you're initializing it with an empty dictionary further up. don't think there is a significant difference between creating a new list vs clearing the old one either

Reply:
> don't think this should ever be null since you're initializing it with an empty dictionary further up. don't think there is a significant difference between creating a new list vs clearing the old one either

It seems that fields added in patch_MapData are not initialized, even if you assign values to them when you define them

Reply:
correct, when i didn't have this null check it wasn't initialised
personally i prefer clearing the old dictionary over recreating it as it's more obvious what the intent is and also possibly more efficient (no new allocation/garbage collection needed)

Reply:
Oh huh, yeah apparently field initialization doesn't work in patch classes in general due to how C# handles it. And that's fair, I generally try to keep things concise/consistent so I was comparing to the other lists that get initialized